_pytest/test_components.py

```diff
@@ -31,9 +24,9 @@ def test_all_components_in_mode_templates_exist(pipeline_template):
 
 
 @pytest.mark.parametrize("component_class", registry.component_classes)
-def test_all_arguments_can_be_satisfied_during_init(component_class):
+def test_all_arguments_can_be_satisfied_during_init(component_class, default_config, component_builder):
```

Comment:
added a couple more comments, but the bulk of the explanation should be done in the docs. It would be really cool if you change the pipeline documentation explanation, adding things that are missing, and marking areas where things are unclear.